hazelcast/src/main/java/com/hazelcast/client/impl/ClientEngineImpl.java

```diff
@@ -375,6 +376,8 @@ public void run() {
                 } else {
                     handleAuthenticationFailure(endpoint, request);
                 }
+            } catch (QueryResultSizeExceededException e) {
```

Comment:
In one of my tests I saw the `QueryResultSizeExceededException` being logged by that catch. I'll have a another look which test that was.